### PR TITLE
Feat: History Management

### PR DESCRIPTION
**Pull request by bharati-21**
This PR contains the following-
- Authenticated users can watch a video which gets added to the history
- Users can clear their watch history
- Users can also delete a previously watched video from their watch history
- If Users watch an previously watched video, history gets updated with the latest video shown at the top.

### CODE REVIEW

**Review by simran61 — APPROVED** [review]
Nice UI
Clean code and folder structure

File: src/components/PlaylistModal/index.jsx [hunk 1]
```diff
@@ -10,15 +10,27 @@ import { PlaylistOption } from "./PlaylistOption";
 const PlaylistModal = ({ video, setShowPlaylistModal }) => {
 	const { theme } = useTheme();
 	const { authToken } = useAuth();
-	const { userDataDispatch, playlists } = useUserData();
+	const { userDataDispatch, playlists, userDataLoading } = useUserData();
 	const { showToast } = useToast();
```
Comment (MonikaJhingan) [inline src/components/PlaylistModal/index.jsx:14]: Use of useAuth() and useToast()  has been nicely done.

File: src/contexts/user-data-context.js [hunk 2]
```diff
@@ -108,6 +109,32 @@ const UserDataProvider = ({ children }) => {
 		}
 	};
 
+	const callGetHistoryVideosService = async () => {
+		try {
+			const {
+				data: { history },
+			} = await getHistoryVideos(authToken);
+			userDataDispatch({
```
Comment (MonikaJhingan) [inline src/contexts/user-data-context.js:117]: async and await are properly used.

File: src/pages/Explore/explore.css [hunk 3]
```diff
@@ -3,7 +3,8 @@
 .playlists-main .container,
 .playlists-videos-list-main .container,
 .video-main .container,
-.playlists-videos-list-main .container {
+.playlists-videos-list-main .container,
+.history-main .container {
     height: 100%;
```
Comment (MonikaJhingan) [inline src/pages/Explore/explore.css:8]: Please use a consistent unit. If using rem then pls try to be consistent with it.

File: src/pages/Explore/explore.css [hunk 4]
```diff
@@ -3,7 +3,8 @@
...
 	gap: var(--space-3);
 	padding: var(--space-3) var(--space-1);
```
Comment (MonikaJhingan) [inline src/pages/Explore/explore.css:10]: Use of var has been carefully done which is good.

File: src/pages/History/index.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+							</Link>
+						</div>
+					)}
+				</div>
+			)}
+		</main>
+	);
+};
+
+export { History };
```
Comment (MonikaJhingan) [inline src/pages/History/index.jsx:86]: The code is very well indented just a little more space is there at the left side.

File: src/routes/NavRoutes.jsx [hunk 6]
```diff
@@ -12,6 +12,7 @@ import {
 	Playlists,
 	PlaylistVideos,
 	SingleVideo,
+	History,
```
Comment (MonikaJhingan) [inline src/routes/NavRoutes.jsx:15]: A very neat display of importing pages.

File: src/components/PlaylistModal/PlaylistOption/index.jsx [hunk 7]
```diff
@@ -64,8 +80,15 @@ const PlaylistOption = ({ video, playlist }) => {
 		} catch (error) {
 			showToast("Failed to delete playlist.", "error");
 		}
+
```
Comment (MonikaJhingan) [inline src/components/PlaylistModal/PlaylistOption/index.jsx:83]: Use of try  catch block is extensively done which is actually very nice for error handling.

File: src/backend/controllers/HistoryController.js [hunk 8]
```diff
@@ -54,16 +54,16 @@ export const addVideoToHistoryHandler = function (schema, request) {
...
-      return new Response(
-        409,
-        {},
-        {
-          errors: ["The video is already in your history"],
-        }
-      );
+        const filteredHistory = user.history.filter((item) => item._id !== videoId);
+        const updatedHistory = [{...video}, ...filteredHistory];
+        this.db.users.update({ history: updatedHistory });
```
Comment (MonikaJhingan) [inline src/backend/controllers/HistoryController.js:62]: Updating the history functionality is well implemented.

**Review by MonikaJhingan — APPROVED** [review]
The overall logic and the ui looks amazingly good. Need to be merged.

**Comment by bharati-21** [comment]
> Nice UI Clean code and folder structure

Thank you for your review.

File: src/pages/Explore/explore.css (hunk 3 above)
Comment (bharati-21) [reply src/pages/Explore/explore.css:8]: I wanted the child container to occupy the entire space that's available relative to the parent. Hence 100%.

**Comment by bharati-21** [comment]
> The overall logic and the ui looks amazingly good. Need to be merged.

Thank you for your review.

File: src/pages/History/index.jsx (hunk 5 above)
Comment (bharati-21) [reply src/pages/History/index.jsx:86]: Sure, I shall use code formatter to format the code once again.

**Pull request merged**